contractcourt/channel_arbitrator.go

```diff
@@ -244,12 +259,45 @@ func (c *ChannelArbitrator) Start() error {
 		return err
 	}
 
+	// Check the database for the close status of this channel.
+	isClosing, closeHeight, closeType, err := c.cfg.IsPendingClose()
+	if err != nil {
+		c.cfg.BlockEpochs.Cancel()
+		return err
+	}
+
+	// If the channel has been marked pending close in the database, and we
+	// haven't transitioned the state machine to StateContractClosed (or a
+	// suceeding state), then a state transition most likely failed. We'll
+	// try to recover from this by manually advancing the state by setting
+	// the corresponding close trigger.
+	trigger := chainTrigger
+	triggerHeight := uint32(bestHeight)
+	if isClosing {
+		switch c.state {
+		case StateDefault:
+			fallthrough
+		case StateBroadcastCommit:
+			fallthrough
+		case StateCommitmentBroadcasted:
+			switch closeType {
+			case channeldb.LocalForceClose:
+				trigger = localCloseTrigger
+			case channeldb.RemoteForceClose:
+				trigger = remoteCloseTrigger
+			}
+			triggerHeight = closeHeight
```

Comment:
👍 

This should actually resolve a known bug in the case of neutrino.

Reply:
would we start searching from most recent block height instead?

Reply:
As in, if best height is 10, but then the event happened at height 7, we would set the wrong height hint.

Reply:
didn't know that was an existing issue, great that we have a fix 🎉